stack-graphs/src/filtered_stack_graph.rs

```diff
@@ -0,0 +1,386 @@
+use super::Filter;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Debug)]
+pub struct StackGraph {
+    files: Files,
+    nodes: Nodes,
+    edges: Edges,
+}
+
+impl StackGraph {
+    pub fn reconstruct(&self) -> crate::graph::StackGraph {
+        todo!()
+    }
```

Comment:
The pattern we usually use here is to define something like the following:
```rust
   fn load_into(&self, graph: &mut crate::graph::StackGraph) {
       // loop over self.files and add those to graph
       // loop over self.nodes and add those to graph
       // loop over self.edges and add those to graph
   }
```

This gives the user more control over which instance the data is loaded into, instead of always doing the allocation (and then maybe later copying).

If the elements are added in this order, everything should be available. If a file already exists in graph, we probably want to return an error, to prevent accidental merging of graphs.